### PR TITLE
Login Url Context Mixin

### PR DESCRIPTION
**Pull request by besseym**
Login Url Context Mixin to get the user back to the url they were trying to access before they authenticated with Okta.

### CODE REVIEW

File: mmvizutil/security/access.py [hunk 1]
```diff
@@ -41,3 +42,19 @@ def dispatch(self, request, *args, **kwargs):
...
         return super().dispatch(request, *args, **kwargs)
+
+
+class LogInUrlMixin(ContextMixin):
+    """Sets the login url on itself and in the context"""
+    def __init__(self):
+        self.login_url = None
+
+    def dispatch(self, request, *args, **kwargs):
+        state = self.request.build_absolute_uri()
```
Comment (kanastasakis26) [inline mmvizutil/security/access.py]: Should we think of state as a dictionary or a delimited string? 
Doing so would make it easier to add other state variables.
ex:
For variables
has_candy=1
landing_page=https://gumdrops.com

The state could be 
state=urlencode(has_candy=1&landing_page=https://gumdrops.com)
OR 
state=urlencode({'has_candy':1,'landing_page':'https://gumdrops.com'})

Comment (besseym) [reply mmvizutil/security/access.py]: Unfortunately, this appears to be not so straight forward...

The result of 
`state=urlencode({'has_candy':1,'landing_page':'https://gumdrops.com'})`
is
`state=has_candy=1&landing_page=https%3A%2F%2Fgumdrops.com`

and the result of
`urlencode("'has_candy'=1&'landing_page'='https://gumdrops.com'")`
is
`TypeError: not a valid non-string sequence or mapping object`

Comment (besseym) [reply mmvizutil/security/access.py]: Also, the result of 
`urlencode({'has_candy':1,'landing_page':'https://gumdrops.com'}.__repr__())`
is
`TypeError: not a valid non-string sequence or mapping object`

Comment (besseym) [reply mmvizutil/security/access.py]: and pickle.dumps doesn't work either

Comment (kanastasakis26) [reply mmvizutil/security/access.py]: Oh well.

Comment (besseym) [reply mmvizutil/security/access.py]: I'll merge it in as is for now. We can expand it in the future once we have a need and figure out how to encode several variables as a value for the state parameter.

**Pull request merged**